### PR TITLE
Fixes color picker in code editor - it now only changes one color

### LINKED ISSUE #29100: Color Picker affects multiple colors
**Godot version:**
3.1.1

**Issue description:**
When having a list of Colors, using Color Picker on one of them affects each color in the list.

**Steps to reproduce:**
grafik

### PR DESCRIPTION
**Pull request by iwek7**
Closes #29100
This PR fixes multiple issues with color picker that can be used in code editor. 

### Before:
old_picker
Notice that all matching arguments list in line are changing (also function argument list not only colors!) and there is delay before change.

### After:
correct_picker

### Fixes include:
1) Now only argument list of color under cursor changes
2) Now values update without delay
3) ColorPicker is correctly (I believe) set to deffered mode so change signal function gets called only when user actually selects color and not at every mouse movement - this can update performance quite a bit

### Not fixed:
Changes done via color picker in this way can't be undone, will create issue for that.

### CODE REVIEW

File: editor/plugins/script_text_editor.cpp [hunk 1]
```diff
@@ -1612,10 +1614,20 @@ void ScriptTextEditor::_color_changed(const Color &p_color) {
...
-	String new_line = line.replace(color_args, new_args);
+	String line_with_replaced_args = _create_line_with_replaced_color_args(new_args);
+
 	color_args = new_args;
-	code_editor->get_text_edit()->set_line(color_line, new_line);
+	code_editor->get_text_edit()->set_line(color_position.x, line_with_replaced_args);
+	code_editor->get_text_edit()->update();
+}
+
+String ScriptTextEditor::_create_line_with_replaced_color_args(String new_args) {
```
Comment (akien-mga) [inline editor/plugins/script_text_editor.cpp]: I don't think this needs to be a separate method if it's only used in one place. If the code needs more documentation, you can add a comment to say that it "creates line with replaced color args" directly.

**Comment by akien-mga** [comment]
Note that the emails you used to sign the commit (different for authoring it and committing it apparently) are not registered as linked to your GitHub account, so the commits don't appear as authored by your "@iwek7" user account.

It's not a problem for Git, but if you want to claim the commit back, you can add these emails as secondary emails in your GitHub settings.

File: editor/plugins/script_text_editor.cpp (hunk 1 above)
Comment (akien-mga) [reply editor/plugins/script_text_editor.cpp]: Otherwise the rest seems good, so it should be good to merge once that's fixed.

**Review by akien-mga — APPROVED** [review]
Marking as approved so that I remember to merge it once rebased to take into account above comments.

**Comment by iwek7** [comment]
@akien-mga I've made requested change but I don't agree with it :) - what really gives us saving on private methods? It is good when method does one thing only and having methods as small as possible is usually considered good practice.

Regarding wrong commit author thanks for bringing this up, I've amended commit author and I hope it is correct now, could you confirm or guide me where to see wrong email so I can fix it? :) I'd like to be able to claim this commit without adding secondary mail.

Thanks for review!

File: editor/plugins/script_text_editor.cpp (hunk 1 above)
Comment (iwek7) [reply editor/plugins/script_text_editor.cpp]: method removed

**Comment by akien-mga** [comment]
> what really gives us saving on private methods? It is good when method does one thing only and having methods as small as possible is usually considered good practice.

Godot is a huge codebase, with pretty big classes. Adding private methods everywhere to encapsulate a handful of lines of code that are meant to be used only in one method makes code harder to read and refactor. In this single class if we applied that logic we might easily get 50 or 100 private methods more than what it currently has, for little gain apart from growing the size of the header.

> Regarding wrong commit author thanks for bringing this up, I've amended commit author and I hope it is correct now, could you confirm or guide me where to see wrong email so I can fix it? :) I'd like to be able to claim this commit without adding secondary mail.

The commit author is now using your GitHub email, but your committer identity is still a different one, as seen here:
Screenshot_20190808_171918

Use `git show --pretty=full` to see author and committer details. Most likely your committer identity is the one configured in `~/.gitconfig` or `.git/config`.

**Comment by iwek7** [comment]
I think I've fixed commit ownership issue now by changing mail, but if it is still not ok just merge it please.

> Godot is a huge codebase, with pretty big classes. Adding private methods everywhere to encapsulate a handful of lines of code that are meant to be used only in one method makes code harder to read and refactor. In this single class if we applied that logic we might easily get 50 or 100 private methods more than what it currently has, for little gain apart from growing the size of the header.

Godot is codebase is not THAT big and pushing for smaller classes and methods is exactly what companies with huge codebases are doing more often than not. But I suppose we could discuss this issue for a long time without much merit - in the end it's your codebase and your rules to enforce. Let's just fix this colorpicker already :).

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.

**Pull request merged**